src/test/java/org/agbrothers/englishbot/service/LessonServiceTest.java

```diff
@@ -10,96 +12,77 @@
 import static org.mockito.Mockito.*;
 
 class LessonServiceTest {
+    final LessonRepository lessonRepository = mock(LessonRepository.class);
     final WordPoolService wordPoolService = mock(WordPoolService.class);
-    final LessonService lessonService = new LessonService(wordPoolService);
+    final LessonService lessonService = new LessonService(wordPoolService, lessonRepository);
 
     @Test
     void getLesson_LessonWithCurrentWordExists_ExpectNoLessonCreation() {
         List<Word> words = Collections.singletonList(new Word());
-        Lesson lesson = new Lesson(words, words);
+        User user = new User();
+        Lesson lesson = new Lesson(words, words, user);
         lesson.setCurrentWord(new Word());
-        String chatId = "testChatId";
-        Map<String, Lesson> lessonRegistry = new HashMap<>();
-        lessonRegistry.put(chatId, lesson);
-        lessonService.setLessonRegistry(lessonRegistry);
 
-        Lesson result = lessonService.getLesson(chatId);
+        when(lessonRepository.findLessonByUser(user)).thenReturn(lesson);
+
+        lessonService.getLesson(user);
 
-        assertEquals(lesson, result);
         verify(wordPoolService, never()).getRandomWordPool();
     }
 
     @Test
     void getLesson_NoLesson_ExpectNewLesson() {
-        String chatId = "testChatId";
-        Map<String, Lesson> lessonRegistry = new HashMap<>();
-        lessonService.setLessonRegistry(lessonRegistry);
-        List<Word> words = Collections.singletonList(new Word());
+        User user = new User();
+        when(lessonRepository.findLessonByUser(user)).thenReturn(null);
 
-        when(wordPoolService.getRandomWordPool()).thenReturn(words);
+        lessonService.getLesson(user);
 
-        Lesson result = lessonService.getLesson(chatId);
-
-        assertNotNull(result);
-        assertEquals(result, lessonRegistry.get(chatId));
+        verify(lessonRepository, times(1)).saveAndFlush(any(Lesson.class));
```

Comment:
remove a redundant second parameter in `verify` method

Reply:
Thanks for pointing this out.